src/meta/Intersection.ts

```diff
@@ -0,0 +1,89 @@
+import global from '@dojo/shim/global';
+import WeakMap from '@dojo/shim/WeakMap';
+import Map from '@dojo/shim/Map';
+import { createHandle } from '@dojo/core/lang';
+import { Base } from './Base';
+
+import 'intersection-observer';
+
+interface ExtendedIntersectionObserverEntry extends IntersectionObserverEntry {
+	readonly isIntersecting: boolean;
+}
+
+interface IntersectionDetail extends IntersectionGetOptions {
+	entries: WeakMap<Element, IntersectionResult>;
+	observer: IntersectionObserver;
+}
+
+export interface IntersectionGetOptions {
+	root?: string;
+	rootMargin?: string;
+	threshold?: number[];
+}
+
+export interface IntersectionResult {
+	intersectionRatio: number;
+	isIntersecting: boolean;
+}
+
+const defaultIntersection: IntersectionResult = Object.freeze({
+	intersectionRatio: 0,
+	isIntersecting: false
+});
+
+export class Intersection extends Base {
+	private readonly _details: Map<string, IntersectionDetail> = new Map();
```

Comment:
Small nit, but you could `private readonly _details = new Map<string, IntersectionDetail>();`